### PR TITLE
Added better examples to the fcrepo-camel component

### PR DESCRIPTION
**Pull request by acoburn**
This addresses https://jira.duraspace.org/browse/FCREPO-1282
(more examples would be a good thing)

added OSGi example
added scala-based example

### CODE REVIEW

File: examples/fcrepo-camel-osgi/src/main/resources/OSGI-INF/blueprint/camel-context.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    <property name="authPassword" value="${fcrepo.authPassword}"/>
+    <property name="authHost" value="${fcrepo.authHost}"/>
+    <property name="secure" value="${fcrepo.secure}"/>
+  </bean>
+
+  <camelContext xmlns="http://camel.apache.org/schema/blueprint">
+    <package>org.fcrepo.camel.examples.osgi</package>
+
+    <!-- Configuration of REST indexing endpoint -->
+    <restConfiguration component="restlet" port="{{rest.port}}"/>
```
Comment (awoods) [inline examples/fcrepo-camel-osgi/src/main/resources/OSGI-INF/blueprint/camel-context.xml]: Does the `{{...}}` syntax also get replaced by property values in a similar fashion to `${...}`? or does it have some other purpose?

File: examples/solr-indexer/pom.xml [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+  <build>
+    <defaultGoal>install</defaultGoal>
+
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>2.5.1</version>
+        <configuration>
+          <source>1.6</source>
```
Comment (awoods) [inline examples/solr-indexer/pom.xml]: Any reason not to at least use `source` and `target` values of `1.7`?

**Comment by awoods** [comment]
What are the main differences between the `solr-indexer` and `fcrepo-camel-osgi` example projects? When would someone reference one over the other?

File: examples/solr-indexer/pom.xml (hunk 2 above)
Comment (acoburn) [inline examples/solr-indexer/pom.xml]: Ah yes, good catch.

File: examples/fcrepo-camel-osgi/src/main/resources/OSGI-INF/blueprint/camel-context.xml (hunk 1 above)
Comment (acoburn) [inline examples/fcrepo-camel-osgi/src/main/resources/OSGI-INF/blueprint/camel-context.xml]: Yes, the {{...}} syntax is for property values. The difference is that you can use {{...}} inside a camel context (i.e. in the URIs) while ${...} can only be used outside the camel context (i.e. in a bean definition) -- the property values, however, can all be defined in a single configuration file. The documentation for this is here: http://camel.apache.org/using-propertyplaceholder.html

**Comment by acoburn** [comment]
My mistake, the solr-indexer example wasn't supposed to be part of the commit -- it is a half-baked version of fcrepo-camel-osgi

File: examples/fcrepo-camel-osgi/src/main/resources/OSGI-INF/blueprint/camel-context.xml (hunk 1 above)
Comment (acoburn) [inline examples/fcrepo-camel-osgi/src/main/resources/OSGI-INF/blueprint/camel-context.xml]: tl;dr 
the URIs in camel are the main way to configure endpoints, and the camel folks invented a syntax for accessing certain things like headers, exceptions, the body, etc within those URIs like so: ${headers}

This gets in the way of properties which are typically defined in the same way, so this is a means for accessing those, too.

**Comment by awoods** [comment]
Resolved with: https://github.com/fcrepo4/fcrepo-camel/commit/0dbe6b6265b3d9012fdcbe35cbfef50c65d6408b

**Pull request closed without merging**